scarb/tests/proc_macro_server.rs

```diff
@@ -0,0 +1,45 @@
+use assert_fs::prelude::PathChild;
+use assert_fs::TempDir;
+use proc_macro_server_helpers::TestProcMacroServer;
+use proc_macro_server_helpers::SIMPLE_MACROS;
+use scarb_proc_macro_server_types::methods::defined_macros::DefinedMacros;
+use scarb_proc_macro_server_types::methods::defined_macros::DefinedMacrosParams;
+use scarb_test_support::cairo_plugin_project_builder::CairoPluginProjectBuilder;
+use scarb_test_support::project_builder::ProjectBuilder;
+
+mod proc_macro_server_helpers;
+
+#[test]
+fn defined_macros() {
+    let t = TempDir::new().unwrap();
+    let plugin_package = t.child("some");
+
+    CairoPluginProjectBuilder::default()
+        .lib_rs(SIMPLE_MACROS)
+        .build(&plugin_package);
+
+    let project = t.child("test_package");
+
+    ProjectBuilder::start()
+        .name("test_package")
+        .version("1.0.0")
+        .lib_cairo(
+            r"
+                #[some]
+                fn main() -> felt252 { 12 }
+            ",
+        )
+        .dep("some", plugin_package)
+        .build(&project);
+
+    let mut proc_macro_server = TestProcMacroServer::new(&project);
+
+    let request = proc_macro_server.request::<DefinedMacros>(DefinedMacrosParams {});
+
+    let response = proc_macro_server.wait_for_response(request).ok();
```

Comment:
sounds like chaining these two methods will be a common pattern? perhaps you could add a shortcut for it in `TestProcMacroServer` ?